lib/pages/league/widgets/dialogs/rules_dialog.dart

```diff
@@ -0,0 +1,384 @@
+import 'package:ax_dapp/service/custom_styles.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class LeagueDialog extends StatefulWidget {
+  const LeagueDialog({super.key});
+
+  @override
+  State<LeagueDialog> createState() => _LeagueDialog();
+}
+
+class _LeagueDialog extends State<LeagueDialog> {
+  final TextEditingController leagueNameController = TextEditingController();
+  final TextEditingController startDateController = TextEditingController();
+  final TextEditingController endDateController = TextEditingController();
+  final TextEditingController teamSizeController = TextEditingController();
+  final TextEditingController participantsController = TextEditingController();
+  final TextEditingController entryFeeController = TextEditingController();
+  final List<bool> _private = <bool>[true, false];
+  final List<bool> _lock = <bool>[true, false];
+  bool vertical = false;
+
+  @override
+  void initState() {
+    super.initState();
+    leagueNameController.addListener(() {
+      final text = leagueNameController.text.toLowerCase();
+      leagueNameController.value = leagueNameController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+      startDateController.value = startDateController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+      endDateController.value = endDateController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+      teamSizeController.value = teamSizeController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+      participantsController.value = participantsController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+      entryFeeController.value = entryFeeController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+  }
```

Comment:
Everything else looks good to me. If you can remove this since we will not be needing it then you are good to go.